core/templates/domain/admin/admin-backend-api.service.ts

```diff
@@ -108,30 +108,34 @@ export class AdminBackendApiService {
     private topicSummaryObjectFactory: TopicSummaryObjectFactory) {}
 
   getData(): Promise<AdminPageData> {
-    return this.http.get<AdminPageDataBackendDict>(
-      AdminPageConstants.ADMIN_HANDLER_URL).toPromise().then(response => {
-      return {
-        demoExplorations: response.demo_explorations,
-        demoCollections: response.demo_collections,
-        demoExplorationIds: response.demo_exploration_ids,
-        oneOffJobStatusSummaries: response.one_off_job_status_summaries.map(
-          this.jobStatusSummaryObjectFactory.createFromBackendDict),
-        humanReadableCurrentTime: response.human_readable_current_time,
-        auditJobStatusSummaries: response.audit_job_status_summaries.map(
-          this.jobStatusSummaryObjectFactory.createFromBackendDict),
-        updatableRoles: response.updatable_roles,
-        roleGraphData: response.role_graph_data,
-        configProperties: response.config_properties,
-        viewableRoles: response.viewable_roles,
-        unfinishedJobData: response.unfinished_job_data.map(
-          this.jobDataObjectFactory.createFromBackendDict),
-        recentJobData: response.recent_job_data.map(
-          this.jobDataObjectFactory.createFromBackendDict),
-        continuousComputationsData: response.continuous_computations_data.map(
-          this.computationDataObjectFactory.createFromBackendDict),
-        topicSummaries: response.topic_summaries.map(
-          this.topicSummaryObjectFactory.createFromBackendDict)
-      };
+    return new Promise((resolve, reject) => {
```

Comment:
Should resolve, reject have types?

Also, what is the general plan for ensuring that everything is fully typed?

Reply:
They have types already. (They are inferred automatically by the ts compiler)

Reply:
Ah ok, thanks. One question to clarify then: does the TS compiler complain if anything isn't fully typed?

Also can you add some info about this "inferring" stuff to the wiki page? I think contributors need to understand the cases in which we need to add types and those in which it's not necessary to do so.

Reply:
> Ah ok, thanks. One question to clarify then: does the TS compiler complain if anything isn't fully typed?

Not yet, but we can enable a check after the angular migration. (Because then everything will have types.)

> Also can you add some info about this "inferring" stuff to the wiki page? I think contributors need to understand the cases in which we need to add types and those in which it's not necessary to do so.

Okay.

Reply:
Why do you need to wait for the Angular migration for everything to have types?

Can't things have types before the Angular migration? We can't really consider types to be fully in place until types are enforced universally, and the Angular migration is going to take a while.

Reply:
Because the structure of the angular js files does not favor typing. There are no classes. But a single 
```typescript
angular.module('oppia', ..., function(...) {  })
```
that contains all the stuff.

So, defining types for angular js files is not possible. I even discussed this thing in my GSoC Proposal(https://docs.google.com/document/d/1fIoXVtO3Br2BtuR2B8XnsmJsHTAd3SjsGhUbjENJuPc/edit#bookmark=id.qceznp891cbq).

Reply:
OK, noted. Thanks (and sorry for forgetting that!)

That said -- a different question. Can the TS compiler complain if anything isn't fully typed, for migrated files only?

Reply:
Nope because it there is no check for that because there is no way it can know if the file is in angular and angular js.

Reply:
OK, thanks. Let's move this to an email thread to see what's possible (it won't block the merge of this PR).